### PR TITLE
SL-615: Search for existing child

### PR DESCRIPTION
**Pull request by cioan**
This allows the user to search for an existing child and to create a new mother-child relationship:

Screenshot 2024-05-20 at 1 03 08 PM

Screenshot 2024-05-20 at 1 02 51 PM

Screenshot 2024-05-20 at 1 03 31 PM

### CODE REVIEW

File: omod/src/main/webapp/pages/children/findChildren.gsp [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            actions: {
+                confirm: function() {
+                    jq.ajax({
+                        url: apiBaseUrl + "/relationship",
+                        type: "POST",
+                        dataType: "json",
+                        contentType: "application/json",
+                        data: dataJson,
+                        success: function( data ) {
+                            emr.successMessage("Relationship has been created");
```
Comment (mogoodrich) [inline omod/src/main/webapp/pages/children/findChildren.gsp:58]: Localization?

File: omod/src/main/webapp/pages/children/findChildren.gsp [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                            addChildDialog.close();
+                            jq("#childName").text("");
+                            setTimeout(navigateBackToChildren, 1000);
+                        }
+                    }).fail(function (data) {
+                        emr.errorMessage("Failed to create relationship: " + data.responseText);
```
Comment (mogoodrich) [inline omod/src/main/webapp/pages/children/findChildren.gsp:64]: Localization?

**Review by mogoodrich — APPROVED** [review]
Generally looks good @cioan , though I added a couple comments about localizing success/error messages.

I also wonder, what happens if you try to add a child to a mother and that association already exists?  Would be good to just make sure it gives sensical error message, or if ot, add one.

**Comment by cioan** [comment]
Thanks @mogoodrich ! I have added localization to those toast messages. Unfortunately, the API allows the creation of duplicate relationships:
Screenshot 2024-05-20 at 2 07 12 PM

I think we have the following options:
1. Modify the API to prevent the creation of duplicate relationships
2. Modify the  [patientSearchWidget](https://github.com/openmrs/openmrs-module-coreapps/blob/master/omod/src/main/webapp/fragments/patientsearch/patientSearchWidget.gsp) to support a configuration filter of the results
3. Write a new function that searches patients who are not already in a given relationship type with the mother
4. Other ideas?

**Comment by cioan** [comment]
Thanks @mogoodrich ! I will add the new generated tickets to the JIRA entry for this PR.

**Pull request merged**